src/MCPClient/lib/clientScripts/move_or_merge.py

```diff
@@ -1,9 +1,72 @@
 #!/usr/bin/env python2
 
+import errno
+import filecmp
 import os
 import shutil
 
 
+def mkdir_p(path):
+    """Create a directory if it doesn't already exist."""
+    try:
+        os.makedirs(path)
+    except OSError as err:
+        if err.errno == errno.EEXIST:
+            pass
+        else:
+            raise
+
+
+def _move_file(src, dst):
+    """
+    Move an individual file from ``src`` to ``dst``.
+    """
+    # Ensure the destination directory exists
+    mkdir_p(os.path.dirname(dst))
+
+    # If the file already exists at the destination, check if it's the same.
+    # If so, we can clean up the original file and we're done.  If not, we're
+    # at risk of losing data, so error out.
+    if os.path.isfile(dst):
+        if filecmp.cmp(src, dst):
```

Comment:
I'm not sure. It does seem very much an edge case - so  maybe it is worth doing the deep comparison and taking the performance hit. Worst case if they are the same file and it has to compare the whole file (if different, presumably it'll bail out when it finds a difference).

Reply:
Thanks @helenst - let's start safe (`shallow=False`) and then we can walk that back later if it becomes a painfully slow situation! Always a tricky place in a CR this! I'll approve because why not?! And then can you make that change and I'll prepare to merge in the next day now this is milestoned.